src/main/java/core/basesyntax/FileWork.java

```diff
@@ -8,7 +13,39 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
+    private static final String START_OF_WORD = "w";
+    private static final String CHAR_AND_SPACE_ONLY_REGEXP = "[^A-z ]";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        StringBuilder words = new StringBuilder();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while ((line = readStringFromFile(br)) != null) {
+                for (String word : line.toLowerCase().replaceAll(CHAR_AND_SPACE_ONLY_REGEXP, "")
+                        .split(" ")) {
+                    if (word.startsWith(START_OF_WORD)) {
+                        words.append(word).append(" ");
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        if (words.length() != 0) {
+            String[] array = words.toString().split(" ");
+            Arrays.sort(array);
+            return array;
+        } else {
+            return new String[0];
+        }
```

Comment:
do we need this check?

Reply:
Yes, otherwise StringBuilder.toString().split(" ") return {""} and tests fail.